### PR TITLE
Massive dependencies updates round

### PR DESCRIPTION
**Pull request by rock3r**
YOLO, as they say.

 * Update the Android Gradle plugin to 3.0.0-alpha6, with Gradle being bumped up to v4.1-milestone-1
 * We get rid of Retrolambda and use the new built-in Android plugin's desugarer (thanks Bazel, we'll owe you forever)
 * Update Kotlin to 1.1.3, and Detekt to M13 (with updated configuration)
 * All Play Services and Firebase dependencies made it to the latest and greatest, 11.0.2
 * All support libraries are on 25.4.0
 * Static Analysis? All-new Checkstyle 8.0 and PMD 5.8.1 are joined by the most painful 0.0.1 bump ever, Findbugs, which finally makes it to v3.0.1
 * There's a shower of minor and revision build bumps too, for greater justice
 * Now, we have updated all those static analysis tools, of course some new things would pop up, so:
    * Lint pointed out some potential crashes in vectors' paths, and some unused resources, which we have addressed
    * Long Vector Drawable paths are ignored because they seem to work just fine in real-world scenarios, and Renderthread's private API usage warnings are shushed because we're still waiting on [a certain PR](https://github.com/Takhion/renderthread/pull/3) to be merged by @Takhion  :P
    * A new annoying rule from PMD has been shushed because it would mean trading off very minor theoretical perf improvements for readability and sanity
 * The CI script has been updated to collect the new Detekt XML reports, too
 * The script that creates the mock `google-services.json` has been updated to have a formally valid Google App ID so that the plugin's validation won't complain anymore.

### CODE REVIEW

**Review by tiwiz — APPROVED** [review]

File: app/build.gradle [hunk 1]
```diff
@@ -105,60 +104,66 @@ play {
...
-    filters = ".*test.*, .*/resources/.*"
-    output = true
-    report = "$project.projectDir/build/reports"
+    version = "1.0.0.M13"
+
+    profile("main") {
+        input = "$projectDir/src/main/java"
+        config = "$rootDir/team-props/static-analysis/detekt-config.yml"
+        filters = ".*test.*,.*/resources/.*,.*/tmp/.*"
+        output = "$project.projectDir/build/reports/detekt.xml"
```
Comment (fourlastor) [inline app/build.gradle]: why do we use `$project.projectDir` here and `$projectDir` 4 lines above? Is it intended?

File: team-props/scripts/ci-mock-google-services-setup.sh [hunk 2]
```diff
@@ -4,15 +4,15 @@
...
+    "project_number": "012345678912",
     "firebase_url": "https://squanchy-mock.firebaseio.com",
     "project_id": "squanchy-mock",
     "storage_bucket": "squanchy-mock.appspot.com"
   },
   "client": [
     {
       "client_info": {
-        "mobilesdk_app_id": "0",
+        "mobilesdk_app_id": "1:012345678912:android:1337deadbeef",
```
Comment (fourlastor) [inline team-props/scripts/ci-mock-google-services-setup.sh:15]: dead beef? 😂

**Review by fourlastor — CHANGES_REQUESTED** [review]
One question

File: team-props/scripts/ci-mock-google-services-setup.sh (hunk 2 above)
Comment (tiwiz) [reply team-props/scripts/ci-mock-google-services-setup.sh:15]: I'm afraid this has something to do with the issue mentioned in @rock3r tweet - https://twitter.com/seebrock3r/status/885828816840609792 - and should ensure that we provide a valid dummy ID for the CI

Comment (rock3r) [reply team-props/scripts/ci-mock-google-services-setup.sh:15]: That is the reason for this change; `deadbeef` is just a throwback thing in [hexspeak](https://en.wikipedia.org/wiki/Hexspeak):

image

I originally considered `cafebabe` as an homage to the JVM but then PC and all so I was all like whatever, let's go non-vegan instead, just for you @fourlastor <3

File: app/build.gradle (hunk 1 above)
Comment (rock3r) [reply app/build.gradle]: Cause I did not really change this line, but I did the other one, and in this one `project.` is unnecessary :)

Comment (rock3r) [reply app/build.gradle]: e3981a446e449d47a94ed5d1b88e53dae0659ebd

File: team-props/scripts/ci-mock-google-services-setup.sh (hunk 2 above)
Comment (fourlastor) [reply team-props/scripts/ci-mock-google-services-setup.sh:15]: Yeah I know why just found funny the name :P

**Review by fourlastor — APPROVED** [review]

**Pull request merged**